src/OrbitGl/TimeGraph.cpp

```diff
@@ -299,13 +305,41 @@ void TimeGraph::ProcessOrbitFunctionTimer(FunctionInfo::OrbitType type,
       break;
     case FunctionInfo::kOrbitTimerStartAsync:
     case FunctionInfo::kOrbitTimerStopAsync:
-      manual_instrumentation_manager_->ProcessAsyncTimer(timer_info);
+      manual_instrumentation_manager_->ProcessAsyncTimerDeprecated(timer_info);
       break;
     default:
       break;
   }
 }
 
+void TimeGraph::ProcessApiEventTimer(const TimerInfo& timer_info) {
+  orbit_api::Event api_event = ManualInstrumentationManager::ApiEventFromTimerInfo(timer_info);
+  switch (api_event.type) {
+    case orbit_api::kScopeStart:
+    case orbit_api::kScopeStop: {
+      ThreadTrack* track = track_manager_->GetOrCreateThreadTrack(timer_info.thread_id());
+      track->OnTimer(timer_info);
+      break;
+    }
+    case orbit_api::kScopeStartAsync:
+    case orbit_api::kScopeStopAsync:
+      manual_instrumentation_manager_->ProcessAsyncTimer(timer_info);
+      break;
+
+    case orbit_api::kTrackInt:
+    case orbit_api::kTrackInt64:
+    case orbit_api::kTrackUint:
+    case orbit_api::kTrackUint64:
+    case orbit_api::kTrackFloat:
+    case orbit_api::kTrackDouble:
```

Comment:
Use `[[fallthrough]]` where applicable.

Reply:
I think it's not needed here. "A common exception is consecutive case labels without intervening code, in which case no annotation is needed" https://google.github.io/styleguide/cppguide.html#Loops_and_Switch_Statements